interfaces/builtin/utils.go

```diff
@@ -96,3 +97,26 @@ func sanitizeSlotReservedForOSOrApp(iface interfaces.Interface, slot *snap.SlotI
 	}
 	return nil
 }
+
+// determine if permanent slot side is provided by the system
+// on classic system some implicit slots can be provided by system or by
+// application snap e.g. avahi (it can be installed as deb or snap)
+// - slot owned by the system (core,snapd snap)  usually requires no action
+// - slot owned by application snap typically requires rules update
+func isSlotSystemSlot(slot *snap.SlotInfo) bool {
+	if release.OnClassic &&
```

Comment:
Do we actually need this? The function today is used in a way that we can't have "avahi" on !classic, right?

Reply:
we need it to determine how is avahi installed.
Avahi can be on classic installed as deb or as snap. And clients should be able to connect to either one.
Trick is when installing avahi on classic as snap, we need to give it right permissions, but without giving those permissions to core snap in case avahi is installed as deb....
So this function should determine if slot in question is system slot ( avahi installed as deb) or not (avahi installed as snap)
We do not have this case yet, but I can image blues, network manager being another good examples....

Reply:
Yeah, what I mean I that I wonder if we need the explicit check for `release.OnClassic` here? Given the current uses/use-cases for this its fine mostly wondering if at some point down the line we write code like "implicitSystemPermanentSlot(kernelModuleCOntrolSlot)" and wonder why we get different results on core vs classic. Maybe @pedronis has an opinion?